sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -12,19 +12,161 @@
 #include <string>
 
 namespace Azure { namespace Core {
+
+  /**
+   * @brief An exception that gets thrown when @DateTime error occurs.
+   */
+  class DateTimeException : public std::runtime_error {
+  public:
+    /**
+     * @brief Construct with message string.
+     *
+     * @param message Message string.
+     */
+    explicit DateTimeException(std::string const& message) : std::runtime_error(message) {}
+  };
+
   /**
    * @brief Manages date and time in standardized string formats.
    */
   class DateTime {
   public:
-    /// A type that represents tick spans.
-    typedef uint64_t IntervalType;
-
-  private:
-    //  Number of seconds between 01-01-1970 and 01-01-1601.
-    static constexpr IntervalType WindowsToPosixOffsetSeconds = 11644473600LL;
+    /**
+     * @brief Represents time duration.
+     */
+    class Duration {
+      friend class DateTime;
+
+      int64_t m_100nsIntervals;
+      constexpr explicit Duration(int64_t intervalsOf100ns) : m_100nsIntervals(intervalsOf100ns) {}
+
+      static constexpr int64_t NanosecondResolution = 100;
+      static constexpr int64_t IntervalsOf100nsPerMicrosecond = 10;
+      static constexpr int64_t IntervalsOf100nsPerMillisecond
+          = 1000 * IntervalsOf100nsPerMicrosecond;
+      static constexpr int64_t IntervalsOf100nsPerSecond = 1000 * IntervalsOf100nsPerMillisecond;
+      static constexpr int64_t IntervalsOf100nsPerMinute = 60 * IntervalsOf100nsPerSecond;
+      static constexpr int64_t IntervalsOf100nsPerHour = 60 * IntervalsOf100nsPerMinute;
+
+    public:
+      constexpr Duration() : m_100nsIntervals(0) {}
+
+      static constexpr Duration FromHours(int32_t hours)
```

Comment:
Add doxygen docs to all public surface area.

Reply:
I didn't think we were going to have functions like this at all.
I thought we could just set a Duration variable like this: Duration d = 2 * Duration::Hour

And then, instead of "AddHours" function just do:  d += 3 * Duration:Hour

So, I just would have expected a really simple Duration type that wraps a signed int64, overloads a few operators (+, -, ==, !=, <, >, <=, >=) and defines some constants (Second, Minutes, Hour) for whatever units the  int64 is in. Nothing else; no methods.